python/tk_premiere/premiere.py

```diff
@@ -69,6 +70,64 @@ def node_id(self):
+        # <?xpacket begin="﻿" id="W5M0MpCehiHzreSzNTczkc9d"?>
+        # <x:xmpmeta xmlns:x="adobe:ns:meta/" x:xmptk="Adobe XMP Core 7.1-c000 79.b0f8be9, 2021/12/08-19:11:22        ">
+        #    <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
+        #       <rdf:Description rdf:about=""
+        #             xmlns:premierePrivateProjectMetaData="http://ns.adobe.com/premierePrivateProjectMetaData/1.0/">
+        #          <premierePrivateProjectMetaData:Column.Intrinsic.Name>blah</premierePrivateProjectMetaData:Column.Intrinsic.Name>
+        #          <premierePrivateProjectMetaData:Column.PropertyText.Label>BE.Prefs.LabelColors.7</premierePrivateProjectMetaData:Column.PropertyText.Label>
+        #          <premierePrivateProjectMetaData:Column.Intrinsic.MediaType>Bin</premierePrivateProjectMetaData:Column.Intrinsic.MediaType>
+        #          <premierePrivateProjectMetaData:Column.PropertyBool.Good>True</premierePrivateProjectMetaData:Column.PropertyBool.Good>
+        #          <premierePrivateProjectMetaData:Column.PropertyBool.Hide>False</premierePrivateProjectMetaData:Column.PropertyBool.Hide>
+        #          <premierePrivateProjectMetaData:Column.PropertyBool.PropagatedHide>False</premierePrivateProjectMetaData:Column.PropertyBool.PropagatedHide>
+        #          <premierePrivateProjectMetaData:myprop>test</premierePrivateProjectMetaData:myprop>
+        #       </rdf:Description>
+        #    </rdf:RDF>
+        # </x:xmpmeta>
+        # <?xpacket end="w"?>
+        m = re.search(
+            r"<premierePrivateProjectMetaData:%s>(.*)</premierePrivateProjectMetaData:%s>" % (property, property),
+            meta_data
+        )
+        if not m:
+            # Since we don't get entries for properties which have not been set
+            # we can't know if the property exists or not, or is empty.
+            # So we return None for the time being...
+            return None
+        return m.group(1)
+
+    def set_meta_data(self, property, value):
+        """
+        Set the meta data value for the given property.
+
+        :param property: A meta data property name, like 'myproperty' or 'Column.PropertyBool.Hide' for
+                         standard meta data properties.
+        :param value: The value to set.
+        :returns: The value which was actually set.
+        """
+        meta_data = self.item.getProjectMetadata()
+        # Since we pass the name of the property to set, we can just replace any property entry in
+        # the meta data string with what we want to set.
+        repl = re.sub(
+            r"<premierePrivateProjectMetaData:[^<]+>(.*)</premierePrivateProjectMetaData:.*>",
+            r"<premierePrivateProjectMetaData:%s>%s</premierePrivateProjectMetaData:%s>" % (property, value, property),
+            meta_data,
+            1
+        )
+        self.item.setProjectMetadata(repl, [property])
```

Comment:
It would be useful to have a link to this https://ppro-scripting.docsforadobe.dev/item/projectitem.html#projectitem-setprojectmetadata
to see that the second arg is the updated fields. I guess it picks the first one it finds and that's why it works?